webapp/src/main/java/com/box/l10n/mojito/service/asset/VirtualTextUnitBatchUpdaterService.java

```diff
@@ -115,6 +123,10 @@ public void updateTextUnits(
 
     performLeveraging(savedTextUnits, nameToUsedtextUnitDTOs, contentToTextUnitDTOs);
 
+    if (aiTranslationJobScheduler != null) {
+      scheduleAITranslation(savedTextUnits.keySet(), asset.getRepository());
```

Comment:
Should we filter already leveraged TUs after the prior step and only schedule work where strictly needed? (mirroring another comment)